pages/api/recent-posts.ts

```diff
@@ -0,0 +1,110 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+
+import { ExtendedRecordMap } from 'notion-types'
+import {
+  getBlockParentPage,
+  getBlockTitle,
+  getPageProperty,
+  idToUuid
+} from 'notion-utils'
+
+import * as config from '@/lib/config'
+import { getSiteMap } from '@/lib/get-site-map'
+import { getSocialImageUrl } from '@/lib/get-social-image-url'
+import { getCanonicalPageUrl } from '@/lib/map-page-url'
+
+type BlogPost = {
+  title: string
+  description: string
+  published: number // unix timestamp
+  url: string
+  authors: string[]
+  coverImage: {
+    url: string
+    width: number
+    height: number
+  }
+}
+
+type Response = {
+  posts: BlogPost[]
+}
+
+/**
+ * This handler exposes recent blog posts as a JSON object for displaying post
+ * previews on the rest of the site.
+ */
+export default async function (req: NextApiRequest, res: NextApiResponse) {
+  const siteMap = await getSiteMap()
+  const ttl = 24 * 60 * 60 // 24 hours
+  const data: Response = {
+    posts: []
+  }
+
+  const limit = 5
+
+  for (const pagePath of Object.keys(siteMap.canonicalPageMap)) {
+    const pageId = siteMap.canonicalPageMap[pagePath]
```

Comment:
A chunk of this code is duplicated from `feed.tsx`. We could extract it out into common utilities, but I figured a little duplication for these different use-cases is ok.